edgy/__init__.py

```diff
@@ -1,110 +1,177 @@
+
 __version__ = "0.22.0"
+import os
+from importlib import import_module
+from typing import TYPE_CHECKING, Any, NamedTuple, Optional
+
+from monkay import Monkay
 
-from .cli.base import Migrate
-from .conf import settings
-from .conf.global_settings import EdgySettings
-from .core import files
-from .core.connection.database import Database, DatabaseURL
-from .core.connection.registry import Registry
-from .core.db import fields
-from .core.db.constants import CASCADE, RESTRICT, SET_NULL, ConditionalRedirect
-from .core.db.datastructures import Index, UniqueConstraint
-from .core.db.fields import (
-    BigIntegerField,
-    BinaryField,
-    BooleanField,
-    CharField,
-    ChoiceField,
-    CompositeField,
-    ComputedField,
-    DateField,
-    DateTimeField,
-    DecimalField,
-    DurationField,
-    EmailField,
-    ExcludeField,
-    FloatField,
-    IntegerField,
-    JSONField,
-    PasswordField,
-    PlaceholderField,
-    RefForeignKey,
-    SmallIntegerField,
-    TextField,
-    TimeField,
-    URLField,
-    UUIDField,
+from edgy.conf.global_settings import EdgySettings
+from edgy.core.connection import Database, DatabaseURL, Registry
+from edgy.core.db.models import (
+    Manager,
+    Model,
+    ModelRef,
+    RedirectManager,
+    ReflectModel,
+    StrictModel,
 )
-from .core.db.fields.foreign_keys import ForeignKey
-from .core.db.fields.many_to_many import ManyToMany, ManyToManyField
-from .core.db.fields.one_to_one_keys import OneToOne, OneToOneField
-from .core.db.models import Model, ModelRef, ReflectModel, StrictModel
-from .core.db.models.managers import Manager
-from .core.db.querysets import Prefetch, Q, QuerySet, and_, not_, or_
-from .core.extras import EdgyExtra
-from .core.signals import Signal
-from .core.utils.sync import run_sync
-from .exceptions import MultipleObjectsReturned, ObjectNotFound
+from edgy.core.db.querysets import Prefetch, Q, QuerySet, and_, not_, or_
+from edgy.core.utils.sync import run_sync
+
+if TYPE_CHECKING:
+    from .core.db.datastructures import Index, UniqueConstraint
+    from .core.signals import Signal
+    from .exceptions import MultipleObjectsReturned, ObjectNotFound
+
+
+class Instance(NamedTuple):
+    registry: Registry
+    app: Optional[Any] = None
+
 
 __all__ = [
+    "Instance",
+    "get_migration_prepared_registry",
+    "monkay",
     "and_",
     "not_",
     "or_",
     "Q",
+    "EdgyExtra",
+    "EdgySettings",
+    "files",
+    "Migrate",
+    "Prefetch",
+    "QuerySet",
+    "Signal",
+    "settings",
+    "run_sync",
+    # index and constraint
+    "Index",
+    "UniqueConstraint",
+    # some exceptions
+    "MultipleObjectsReturned",
+    "ObjectNotFound",
+    # constants
+    "CASCADE",
+    "RESTRICT",
+    "DO_NOTHING",
+    "SET_NULL",
+    "SET_DEFAULT",
+    "PROTECT",
+    "ConditionalRedirect",
+    # models
+    "ReflectModel",
+    "StrictModel",
+    "Model",
+    "ModelRef",
+    "Manager",
+    "RedirectManager",
+    # fields
```

Comment:
no need for these comments. Lets remove them all

Reply:
I would prefer to order the `__all__`, so we keep track.

Reply:
I have to admit, there is a test for it (test_lazy_imports), so it is not that important anymore.

Reply:
All remains. I'm taking the comments like "#fields"